include/boost/detail/winapi/dll.hpp

```diff
@@ -18,83 +19,186 @@
+    boost::detail::winapi::DWORD_ dwFlags
+);
+
+BOOST_SYMBOL_IMPORT boost::detail::winapi::HMODULE_ WINAPI
 GetModuleHandleW(boost::detail::winapi::LPCWSTR_ lpFileName);
 
+BOOST_SYMBOL_IMPORT boost::detail::winapi::DWORD_ WINAPI
+GetModuleFileNameW(
+    boost::detail::winapi::HMODULE_ hModule,
+    boost::detail::winapi::LPWSTR_ lpFilename,
+    boost::detail::winapi::DWORD_ nSize
+);
+
+
 BOOST_SYMBOL_IMPORT boost::detail::winapi::BOOL_ WINAPI
 FreeLibrary(boost::detail::winapi::HMODULE_ hModule);
 
 BOOST_SYMBOL_IMPORT boost::detail::winapi::FARPROC_ WINAPI
 GetProcAddress(boost::detail::winapi::HMODULE_ hModule, boost::detail::winapi::LPCSTR_ lpProcName);
-}
-#endif
 
-namespace boost {
-namespace detail {
-namespace winapi {
+BOOST_SYMBOL_IMPORT boost::detail::winapi::FARPROC_ WINAPI
+VirtualQuery(
+    boost::detail::winapi::LPCVOID_ lpAddress,
+    boost::detail::winapi::MEMORY_BASIC_INFORMATION_* lpBuffer,
+    boost::detail::winapi::ULONG_PTR_ dwLength
+);
+
+} // extern "C"
+#endif // #if !defined( BOOST_USE_WINDOWS_H )
+
+namespace boost { namespace detail { namespace winapi {
 
 #if defined( BOOST_USE_WINDOWS_H )
 typedef ::FARPROC FARPROC_;
 typedef ::NEARPROC NEARPROC_;
 typedef ::PROC PROC_;
+typedef ::MEMORY_BASIC_INFORMATION MEMORY_BASIC_INFORMATION_;
+
+const DWORD_ DONT_RESOLVE_DLL_REFERENCES_           = DONT_RESOLVE_DLL_REFERENCES;
+const DWORD_ LOAD_IGNORE_CODE_AUTHZ_LEVEL_          = LOAD_IGNORE_CODE_AUTHZ_LEVEL;
+const DWORD_ LOAD_LIBRARY_AS_DATAFILE_              = LOAD_LIBRARY_AS_DATAFILE;
+const DWORD_ LOAD_LIBRARY_AS_DATAFILE_EXCLUSIVE_    = LOAD_LIBRARY_AS_DATAFILE_EXCLUSIVE;
+const DWORD_ LOAD_LIBRARY_AS_IMAGE_RESOURCE_        = LOAD_LIBRARY_AS_IMAGE_RESOURCE;
+const DWORD_ LOAD_WITH_ALTERED_SEARCH_PATH_         = LOAD_WITH_ALTERED_SEARCH_PATH;
 #endif // defined( BOOST_USE_WINDOWS_H )
 
+// We always define the following constants by explicit values, because
+// they may not be defined in windows.h
+//
+// Minimum supported client :  Windows 8
+// Minimum supported server : Windows Server 2012
+//
+// To Windows 7, Windows Server 2008 R2, Windows Vista,
+// and Windows Server 2008: KB2533623 must be installed on the target platform.
+const DWORD_ LOAD_LIBRARY_SEARCH_PLUGIN_DIR_    = 0x00000200;
+const DWORD_ LOAD_LIBRARY_SEARCH_DEFAULT_DIRS_  = 0x00001000;
+const DWORD_ LOAD_LIBRARY_SEARCH_DLL_LOAD_DIR_  = 0x00000100;
+const DWORD_ LOAD_LIBRARY_SEARCH_SYSTEM32_      = 0x00000800;
+const DWORD_ LOAD_LIBRARY_SEARCH_USER_DIRS_     = 0x00000400;
+
 #if !defined( BOOST_NO_ANSI_APIS )
 using ::LoadLibraryA;
+using ::LoadLibraryExA;
 using ::GetModuleHandleA;
-#endif
+using ::GetModuleFileNameA;
+#endif // !defined( BOOST_NO_ANSI_APIS )
 using ::LoadLibraryW;
+using ::LoadLibraryExW;
 using ::GetModuleHandleW;
+using ::GetModuleFileNameW;
 using ::FreeLibrary;
 using ::GetProcAddress;
+using ::VirtualQuery;
 
 #if !defined( BOOST_NO_ANSI_APIS )
-BOOST_FORCEINLINE HMODULE_ load_library(LPCSTR_ lpFileName)
```

Comment:
Please, don't change the coding style of the existing code. I'd rather prefer the new code to follow the existing style.